datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,605 @@
+        del stored_config["checkpoint_file"]
+        if stored_config != loaded_config:
+            logger.debug(
+                f"Found difference in estimator config."
+                f"Estimator loaded from {path} was trained using config: \n"
+                f"{loaded_config}. \nHowever, the current config is: \n"
+                f"{self.config}."
+            )
+        self.model_without_ddp.eval()
+
+    def predict(self, pil_img, box_score_thresh=0.5):
+        """ Get prediction from one image using loaded model
+
+        Args:
+            pil_img (PIL Image): PIL image from dataset.
+            box_score_thresh (float): box score threshold for filter out lower
+            score bounding boxes. Defaults to 0.5.
+
+        Returns:
+            filtered_pred_annotations (List[BBox2D]): high predicted score
+            bboxes from the model.
+        """
+        pil_img = pil_img.unsqueeze(0)
+        img_tensor = pil_img.to(self.device)
+        predicts = self.model_without_ddp(img_tensor)
+        predict_annotations = convert_bboxes2canonical(predicts)[0]
+        filtered_pred_annotations = [
+            box for box in predict_annotations if box.score >= box_score_thresh
+        ]
+        return filtered_pred_annotations
+
+
+def create_dryrun_dataset(config, dataset, split):
+    """
+    create dataset with very small no of images
+    Args:
+        config: (CfgNode): estimator config:
+        dataset: dataset obj must have len and __get_item__
+        split: train, val, test
+
+    Returns dataset: Subset dataset
+
+    """
+    r = np.random.default_rng()
+    indices = r.integers(0, len(dataset), config[split].batch_size * 2)
+    dataset = torch.utils.data.Subset(dataset, indices)
+    return dataset
+
+
+def download_data(config, split):
+    """
+    download dataset from source
+    Args:
+        config: (CfgNode): estimator config:
+        split: train, val, test
+
+    Returns dataset: dataset obj must have len and __get_item__
+
+    """
+    dataset = Dataset.create(
+        config[split].dataset.name,
+        data_root=config.system.data_root,
+        transforms=get_transform(),
+        **config[split].dataset.args,
+    )
+    return dataset
+
+
+def create_sampler(is_distributed, *, dataset, is_train):
+    """ create sample of data
+    Args:
+        is_distributed: whether or not the model is distributed
+        dataset: dataset obj must have len and __get_item__
+        is_train: whether or not the sampler is for training data
+
+    Returns:
+        data_sampler: (torch.utils.data.Sampler)
+
+    """
+    if is_distributed:
+        sampler = torch.utils.data.distributed.DistributedSampler(
+            dataset, shuffle=is_train
+        )
+
+    else:
+        if is_train:
+            sampler = torch.utils.data.RandomSampler(dataset)
+        else:
+            sampler = torch.utils.data.SequentialSampler(dataset)
+    return sampler
+
+
+def load_data(
```

Comment:
How is `load_data` different from `create_dataloader`? Should we rename both methods to improve readability.

Reply:
create_dataloader triggers the data load process, load_data loads the data. We can put in same method but it will bit lengthy. Spliting into two units make more readable I think.

Reply:
Better to rename both methods to be more readable.

Reply:
Agree with @BlairLee